src/App.js

```diff
@@ -10,40 +15,165 @@ class App extends React.Component {
       // Set default value of card deck to new shuffled deck
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
-      currCards: [],
+      round: 0,
+      player1: { winGame: 0, score: 0, result: "", currCard: {} },
+      player2: { winGame: 0, score: 0, result: "", currCard: {} },
+      restart: "",
+      dealButton: <button onClick={this.changeResult}>Deal</button>,
     };
   }
 
   dealCards = () => {
     // this.state.cardDeck.pop() modifies this.state.cardDeck array
-    const newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+    const newCurrCard1 = this.state.cardDeck.pop();
+    const newCurrCard2 = this.state.cardDeck.pop();
+    this.state.player1.currCard = newCurrCard1;
+    this.state.player2.currCard = newCurrCard2;
+  };
+
+  compareCards = () => {
+    if (this.state.player1.currCard.rank === this.state.player2.currCard.rank) {
+      this.state.player1.result = "draw";
+      this.state.player2.result = "draw";
+    } else if (
+      this.state.player1.currCard.rank > this.state.player2.currCard.rank
+    ) {
+      this.state.player1.result = "win";
+      this.state.player1.score += 1;
+      this.state.player2.result = "loss";
+    } else {
+      this.state.player2.result = "win";
+      this.state.player2.score += 1;
+      this.state.player1.result = "loss";
+    }
+  };
+
+  changeResult = () => {
+    this.dealCards();
+    this.compareCards();
     this.setState({
-      currCards: newCurrCards,
+      round: this.state.round + 1,
+      player1: this.state.player1,
+      player2: this.state.player2,
     });
+    if (this.state.cardDeck.length === 0) {
```

Comment:
```suggestion
    if (!this.state.cardDeck.length) {
```

0 is falsy in JavaScript